src/test/java/edu/harvard/iq/dataverse/authorization/users/AuthenticatedUserTest.java

```diff
@@ -320,6 +340,100 @@ public void testHashCode() {
+
+    @Test
+    public void testMutingEmails() {
+        System.out.println("setMutedEmails");
+        testUser.setMutedEmails(mutedTypes);
+        assertEquals(mutedTypes, testUser.getMutedEmails());
+    }
+
+    @Test
+    public void testMutingNotifications() {
+        System.out.println("setMutedNotifications");
+        testUser.setMutedNotifications(mutedTypes);
+        assertEquals(mutedTypes, testUser.getMutedNotifications());
+    }
+
+    @Test
+    public void testMutingInJson() {
+        testUser.setMutedEmails(mutedTypes);
+        testUser.setMutedNotifications(mutedTypes);
+        System.out.println("toJson");
+        JsonObject jObject = testUser.toJson().build();
+
+        Set<String> mutedEmails = new HashSet<>(jObject.getJsonArray("mutedEmails").getValuesAs(JsonString::getString));
+        assertTrue("Set contains two elements", mutedEmails.size() == 2);
+        assertTrue("Set contains REVOKEROLE", mutedEmails.contains("REVOKEROLE"));
+        assertTrue("Set contains ASSIGNROLE", mutedEmails.contains("ASSIGNROLE"));
+
+        Set<String> mutedNotifications = new HashSet<>(jObject.getJsonArray("mutedNotifications").getValuesAs(JsonString::getString));
+        assertTrue("Set contains two elements", mutedNotifications.size() == 2);
+        assertTrue("Set contains REVOKEROLE", mutedNotifications.contains("REVOKEROLE"));
+        assertTrue("Set contains ASSIGNROLE", mutedNotifications.contains("ASSIGNROLE"));
+    }
+
+    @Test
+    public void testHasEmailMuted() {
+        testUser.setMutedEmails(mutedTypes);
+        System.out.println("hasEmailMuted");
+        assertEquals(true, testUser.hasEmailMuted(Type.ASSIGNROLE));
+        assertEquals(true, testUser.hasEmailMuted(Type.REVOKEROLE));
+        assertEquals(false, testUser.hasEmailMuted(Type.CREATEDV));
+        assertEquals(false, testUser.hasEmailMuted(null));
+    }
+
+    @Test
+    public void testHasNotificationsMutedMuted() {
+        testUser.setMutedNotifications(mutedTypes);
+        System.out.println("hasNotificationMuted");
+        assertEquals(true, testUser.hasNotificationMuted(Type.ASSIGNROLE));
+        assertEquals(true, testUser.hasNotificationMuted(Type.REVOKEROLE));
+        assertEquals(false, testUser.hasNotificationMuted(Type.CREATEDV));
+        assertEquals(false, testUser.hasNotificationMuted(null));
+    }
+
+    @Test
+    public void testTypeTokenizer() {
+        final Set<Type> typeSet = Type.tokenizeToSet(
+            Type.toStringValue(
+                Type.tokenizeToSet(" ASSIGNROLE , CREATEDV,REVOKEROLE  ")
+            )
+        );
+        assertTrue("typeSet contains 3 elements", typeSet.size() == 3);
+        assertTrue("typeSet contains ASSIGNROLE", typeSet.contains(Type.ASSIGNROLE));
+        assertTrue("typeSet contains CREATEDV", typeSet.contains(Type.CREATEDV));
+        assertTrue("typeSet contains REVOKEROLE", typeSet.contains(Type.REVOKEROLE));
+    }
+
+    @Test
+    public void testIsEmailMuted() {
```

Comment:
I added this test (and the one below). Hope it helps! Please feel free to expand on them if you like.

Reply:
Thanks, it looks good.